orchagent/routeorch.cpp

```diff
@@ -708,6 +718,15 @@ void RouteOrch::doTask(Consumer& consumer)
                     it_prev++;
             }
         }
+
+        /* Remove next hop group if the reference count decreases to zero */
+        for (auto it_nhg = bulkNhgReducedRefCnt.begin(); it_nhg != bulkNhgReducedRefCnt.end(); it_nhg++)
+        {
+            if (m_syncdNextHopGroups[*it_nhg].ref_count == 0)
+            {
+                removeNextHopGroup(*it_nhg);
```

Comment:
>removeNextHopGroup [](start = 16, length = 18)

Could you write a vs testcase to fail old code and pass new code?

Reply:
need to add a vs test case.

Reply:
A vs test case for this particular issue could be challenging form two aspects:
1. It requires a specific bulk pattern to happen, but we do not have control over how the operations are partition into bulks. Constantly reproducing the issue in vs test could be a problem.
2. The issue is transient, it will get resolved itself and the only trace would be the log messages of failed attempts of nexthop group removals. Basically, there is a limited sign of the issue ever happened.
I think it needs some thinking in developing this test case.

Reply:
i agree with you. I think maybe we need to look into the googletest framework for such sequencing testing. the tests under tests/mock_tests